src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/DataContractJsonSerializer.cs

```diff
@@ -26,7 +26,7 @@ public sealed class DataContractJsonSerializer : XmlObjectSerializer
         private const char LOW_SURROGATE_END = (char)0xdfff;
         private const char MAX_CHAR = (char)0xfffe;
         private const char WHITESPACE = ' ';
-        internal const string SerializerTrimmerWarning = "Json Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the " +
+        internal const string SerializerTrimmerWarning = "Data Contract Serialization and Deserialization might require types that cannot be statically analyzed. Make sure all of the " +
             "required types are preserved.";
```

Comment:
This string won't be localized. Is that okay? Also as this string has now been adopted for use in lots of places and not just for DCJS, it might be better for this to live in a more common class. Maybe DataContract?

Reply:
> This string won't be localized. Is that okay?

Since it gets put on an attribute, it can't be localized. The trimmer only reads the message of RequiresUnreferencedCodeAttribute.

https://github.com/dotnet/runtime/blob/d25620b3fe5fba7b8fd7065f2d5947d0a20c6a30/src/libraries/System.Private.CoreLib/src/System/Diagnostics/CodeAnalysis/RequiresUnreferencedCodeAttribute.cs#L29-L37

Reply:
As @eerhardt points out we can't really localize it so this is the way it is for the rest of the framework. I've moved the warning over to DataContract.